improver_tests/precipitation_type/convection/test_DiagnoseConvectivePrecipitation.py

```diff
@@ -63,6 +59,7 @@ def set_up_precipitation_rate_cube():
     precip_cube = set_up_variable_cube(
         data.astype(np.float32), "lwe_precipitation_rate", "mm h-1", "equalarea",
     )
+    precip_cube = iris.util.squeeze(precip_cube)
```

Comment:
These are very old unit tests, that were using cubes with a realization dimension with only one point in it. I've now demoted this realization dim to a scalar coordinate, removed this extra dim from several unit tests, and added two simple tests for the process method to make sure it can handle multi-realization cubes. Note this class and it's associated tests should be removed after the neighbourhood refactor https://github.com/metoppv/mo-blue-team/issues/214